### DISCUSSION TITLE
Working with Touch + Pen only

### DISCUSSION THREAD

**Discussion by Robert-K** [wojtryb/Shortcut-Composer#53] [Help me]
Hey, first of all, thank you so much! For shortcut-composer, for your theming/customization guides, brushes, everything 💗

I'm working with krita on a convertible latop with a pen, so I have no keyboard while I'm drawing. I'd love to use the pie menus and other features, but the Keyboard Shortcuts settings don't allow me to map them to pen buttons/mouse clicks.
If there's a way to map the pie menus to say right mouse or mouse 4,5 etc, that'd be lovely.

I can also imagine a docker/toolbar with buttons that act much like keys on a keyboard. These could be used as modifier keys (i.e. for scaling, erasing etc or to invoke the pie menus. I'd have my pen in the dominant hand and hold the respective touch key with my non dominant hand.

Do you reckon that's possible? I'm pretty good with python, so I guess I could do some work there, but I'm totally new to Krita scripting.

**Comment by wojtryb** [comment]
Hi!

Mapping mouse buttons directly is out of scope, as krita does not expose `Canvas Inputs` to python, and it would require writing an entire configurable system which allows to remap them.

What I do, is to remap the buttons on the stylus to act as keyboard buttons (this should be possible in the device drivers - I hope your laptop offers that)

For the docker for emulating keyboard with touch - I've never used it but someone recommended https://github.com/Martenfur/TabletFriend. I'd rather look for existing solutions, as this can run in OS, and does not need to be krita related.
